lib/ads.js

```diff
@@ -1401,6 +1401,28 @@ exports.ADSSTATE = {
   RECONFIG:     16,
   STOPPING:     17
 }
+exports.ADSSTATE = ADSSTATE
+
+exports.ADSSTATETXT = {
```

Comment:
Actually we don't need this static thingy, because we can do:

```
const STATE = {
  INVALID:      0,
  IDLE:         1,
  RESET:        2,
  INIT:         3,
  START:        4,
  RUN:          5,
  STOP:         6,
  SAVECFG:      7,
  LOADCFG:      8,
  POWERFAILURE: 9,
  POWERGOOD:    10,
  ERROR:        11,
  SHUTDOWN:     12,
  SUSPEND:      13,
  RESUME:       14,
  CONFIG:       15,
  RECONFIG:     16,
  STOPPING:     17,

  fromId: function(id) {
    return Object.keys(this)[Object.values(this).indexOf(id)]
  }
}
exports.STATE
```

The use is basically:
```
STATE.ERROR -> 11
STATE.fromId(14) -> RESUME
```

Reply:
Object.values is a new feature in ES2017. It is very bleeding edge. Node.js has full support for it from version 7.0.

let me see Object.keys(obj).map

Reply:
it's just one way, there are different similar ways :)

Reply:
changed